internal/provider/resource_archive_file.go

```diff
@@ -0,0 +1,272 @@
+			"output_md5": schema.StringAttribute{
+				Description: "The MD5 checksum of output archive file.",
+				Computed:    true,
+			},
+			"output_file_mode": schema.StringAttribute{
+				Description: "String that specifies the octal file mode for all archived files. For example: `\"0666\"`. " +
+					"Setting this will ensure that cross platform usage of this module will not vary the modes of archived " +
+					"files (and ultimately checksums) resulting in more deterministic behavior.",
+				Optional: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+	}
+}
+
+func (d *archiveFileResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var model fileModel
+	diags := req.Plan.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(updateModel(ctx, &model)...)
+
+	diags = resp.State.Set(ctx, model)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (d *archiveFileResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var model fileModel
+	diags := req.State.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(updateModel(ctx, &model)...)
+
+	diags = resp.State.Set(ctx, model)
+	resp.Diagnostics.Append(diags...)
+}
+
+func updateModel(ctx context.Context, model *fileModel) diag.Diagnostics {
+	var diags diag.Diagnostics
+	outputPath := model.OutputPath.ValueString()
+
+	outputDirectory := path.Dir(outputPath)
+	if outputDirectory != "" {
+		if _, err := os.Stat(outputDirectory); err != nil {
+			if err := os.MkdirAll(outputDirectory, 0755); err != nil {
+				diags.AddError(
+					"Output path error",
+					fmt.Sprintf("error creating output path: %s", err),
+				)
+				return diags
+			}
+		}
+	}
+
+	if err := archive(ctx, *model); err != nil {
+		diags.AddError(
+			"Archive creation error",
+			fmt.Sprintf("error creating archive: %s", err),
+		)
+		return diags
+	}
+
+	// Generate archived file stats
+	fi, err := os.Stat(outputPath)
+	if err != nil {
+		diags.AddError(
+			"Archive output error",
+			fmt.Sprintf("error reading output: %s", err),
+		)
+		return diags
+	}
+
+	sha1, base64sha256, md5, err := genFileShas(outputPath)
+	if err != nil {
+		diags.AddError(
+			"Hash generation error",
+			fmt.Sprintf("error generating hashed: %s", err),
+		)
+		return diags
+	}
+
+	model.OutputSha = types.StringValue(sha1)
+	model.OutputBase64Sha256 = types.StringValue(base64sha256)
+	model.OutputMd5 = types.StringValue(md5)
+	model.OutputSize = types.Int64Value(fi.Size())
+
+	model.ID = types.StringValue(sha1)
+
+	return diags
+}
+
+func (d *archiveFileResource) Update(_ context.Context, _ resource.UpdateRequest, _ *resource.UpdateResponse) {
+}
+
+func (d *archiveFileResource) Delete(_ context.Context, _ resource.DeleteRequest, _ *resource.DeleteResponse) {
+}
```

Comment:
I really wish we could "finish" this resource's implementation and un-deprecate it with deleting the archive file, however that could only be considered in a breaking change for a major version. 🙁